src/gl/gl.cpp

```diff
@@ -181,50 +181,42 @@ void gl_draw_with_lighting(struct indexed_primitive *ip, struct polygon_data *po
 	if (normaldata == nullptr)
 	{
 		static std::vector<struct point3d> normals;
-		static std::vector<int> adjacentTriIndexes;
-		struct point3d normal, e12, e13, triNormal;
+		static point3d zero = { 0.0f, 0.0f, 0.0f };
+		struct point3d e12, e13, triNormal;
+
+		int vertexCount = ip->vertexcount;
+		int indexCount = ip->indexcount;
 
 		normals.clear();
-		normals.shrink_to_fit();
+		normals.resize(vertexCount);
+		std::fill(normals.begin(), normals.end(), zero);
 
 		// Calculate vertex normals by averaging adjacent triangle normals
 		// Vertex normals are calculated here because battle models dont seem to include normals
-		for (uint32_t idx = 0; idx < ip->vertexcount; idx++)
+		for (uint32_t idx = 0; idx < indexCount; idx+=3)
+		{
+			e12 = e13 = triNormal = {0.0f, 0.0f, 0.0f};
```

Comment:
I'd use `zero` as well instead of `{0.0f, 0.0f, 0.0f}`

Reply:
I changed it to `zero`.